### DISCUSSION TITLE
Extract out styles/animations hooks

### DISCUSSION THREAD

**Discussion by jaexplorer** [software-mansion/react-native-reanimated#2411] [Q&A]
Hi, for my components I always follow this folder structure:
MyComponent.tsx
MyComponentStyles.ts
MyComponentConstants.ts
MyComponentUtils.ts

For reanimated hooks such as useSharedValue, useGestureHandler, useAnimationStyles. I like to isolate/decouple out those hooks out of MyComponent.tsx so the file doesn't get so overwhelming with animation logic and simply import instead. What is the best approach for this?

**Comment by jgonet** [comment]
I started creating custom hooks that return needed things (animated style, some shared value, etc).
